### PR TITLE
Cancel docker pull

### LINKED ISSUE #301: [Add BN] Ability to cancel docker pull from loading screen

### PR DESCRIPTION
**Pull request by morrigan**
Closes #301

### CODE REVIEW

**Review by BeroBurny — CHANGES_REQUESTED** [review]
We wanna keep all business logic may as possible in sagas (that mostly reflect on `AddBeaconNode.tsx`)

the simplest soliton (and probably best) can be used in this case is to use the race effect
[https://redux-saga.js.org/docs/api/#raceeffects](https://redux-saga.js.org/docs/api/#raceeffects)
```typescript
export function* pullDockerImage(
    image: string,
): Generator<PutEffect | RaceEffect<CallEffect | TakeEffect>, boolean, [boolean, Action]> {
    yield put(startDockerImagePull());
    const [pull, effect] = yield race([call(BeaconChain.pullImage, image), take(cancelDockerPull)]);
    yield put(endDockerImagePull());

    return effect !== undefined ? false : pull;
}
```
```typescript
export const cancelDockerPull = createAction("network/cancelDockerPull");
```

and this can be simple used in any sagas required to pull an image
**example:**
```typescript
const image = getNetworkConfig(network).dockerConfig.image;
const pullResult = yield call(pullDockerImage, image);
```
**example for cancel:**
```typescript
const onCancelClick = (): void => {
   dispatch(cancelDockerPull());
};
```

**Comment by morrigan** [comment]
@BeroBurny managed to do it somehow but having issues with Saga types (check @ts-ignore).

Also, we need to have 1 function for starting chain. I think `startLocalBeaconSaga` and `startBeaconChainSaga` should unite :) Not sure if left both by mistake.

**Review by BeroBurny — CHANGES_REQUESTED** [review]

File: src/renderer/ducks/beacon/sagas.ts [hunk 1]
```diff
@@ -1,27 +1,51 @@
...
-                    eth1Url,
-                    discoveryPort,
-                    libp2pPort,
-                    rpcPort,
-                }),
-            );
+    meta: {onComplete},
+}: ReturnType<typeof startLocalBeacon>): Generator<CallEffect | PutEffect, void, BeaconChain> {
+    // @ts-ignore
+    const pullSuccess = yield* pullDockerImage(network);
```
[the remark this answers is not in the document]
Comment (morrigan) [reply src/renderer/ducks/beacon/sagas.ts:32]: What's the difference?

[the remark this answers is not in the document]
Comment (BeroBurny) [reply src/renderer/ducks/beacon/sagas.ts:32]: differently is handled by saga middleware

https://redux-saga.js.org/docs/basics/UsingSagaHelpers.html
https://redux-saga.js.org/docs/basics/DeclarativeEffects.html

**Review by BeroBurny — DISMISSED** [review]
:+1:

**Review by BeroBurny — APPROVED** [review]
:+1:

**Pull request merged**